bob/src/cleaner.rs

```diff
@@ -22,27 +24,74 @@ impl Cleaner {
             hard_open_blobs,
             bloom_filter_memory_limit,
             index_memory_limit,
+            index_cleanup_requested: Mutex::new(false),
         }
     }
 
-    pub(crate) fn spawn_task(&self, backend: Arc<Backend>) {
+    pub(crate) fn spawn_task(&self, cleaner: Arc<Cleaner>, backend: Arc<Backend>) {
         tokio::spawn(Self::task(
-            backend,
+            cleaner.clone(),
+            backend.clone(),
             self.old_blobs_check_timeout,
             self.soft_open_blobs,
             self.hard_open_blobs,
             self.bloom_filter_memory_limit,
-            self.index_memory_limit,
         ));
+        tokio::spawn(Self::fast_cleaner_task(
+            cleaner,
+            backend,
+            Duration::from_secs(30),
+            self.index_memory_limit
+        ));
+    }
+
+    pub(crate) async fn request_index_cleanup(&self) {
+        let mut lock = self.index_cleanup_requested.lock().await;
+        *lock = true;
+    }
+
+    async fn fast_cleaner_task(
+        cleaner: Arc<Cleaner>,
+        backend: Arc<Backend>,
+        t: Duration,
+        index_memory_limit: Option<usize>,
+    ) {
+        let mut interval = interval(t);
+        loop {
+            interval.tick().await;
+            if let Some(limit) = index_memory_limit {
+                let mut should_clean_up = false;
+                {
+                    let mut lock = cleaner.index_cleanup_requested.lock().await;
+                    if *lock {
+                        *lock = false;
+                        should_clean_up = true;
+                    }
+                }
+                if should_clean_up {
+                    let mut memory = backend.index_memory().await;
+                    info!("Memory before closing old active blobs: {:?}", memory);
+                    while memory > limit {
+                        if let Some(freed) = backend.close_oldest_active_blob().await {
```

Comment:
As far as I remember, we track the time of the last access to the pearl storage. If this is true, then it would be a good to implement `close_least_recently_used_blob` method and use it here. Closing the least recently used storages is better suited for deletion scenario

Reply:
Additional question:  `close_oldest_active_blob` probably only handles storages with active blob, but indexes loading don't do anything with active blobs, and if I'm right, this function won't offload indexes as expected

Reply:
Now we have separate functions for closing all indexes in least modified holder